eras/shelley/impl/src/Cardano/Ledger/Shelley/TxCert.hs

```diff
@@ -366,7 +366,7 @@ instance
       gen <- decCBOR
       genDeleg <- decCBOR
       vrf <- decCBOR
-      pure (4, GenesisDelegTxCert gen genDeleg vrf)
```

Comment:
Nah, it would work. We'd just have to add a constraint: `ProtVerAtMost era 8`. Which would be fine, but I decided to avoid adding constraints by using the actual type constructor.

Both are equivalent in functionality and safety guarantees.

Reply:
Makes perfect sense, thank you!